kasa/smart/modules/lighttransition.py

```diff
@@ -67,54 +88,90 @@ def _create_features(self):
+        """Update the states."""
+        # Assumes any device with state in sysinfo supports on and off and
+        # has maximum values for both.
+        # v2 adds separate on & off states
+        # v3 adds max_duration except for ks240 which is v2 but supports it
+        if not self._supports_on_and_off:
+            self._enabled = self.data["enable"]
+            return
+
+        if self._state_in_sysinfo:
+            on_max = self._device.sys_info.get(
+                "max_fade_on_time", self.MAXIMUM_DURATION
+            )
+            off_max = self._device.sys_info.get(
+                "max_fade_off_time", self.MAXIMUM_DURATION
+            )
+            on_enabled = bool(self._device.sys_info["gradually_on_mode"])
+            off_enabled = bool(self._device.sys_info["gradually_off_mode"])
+            on_duration = self._device.sys_info["fade_on_time"]
+            off_duration = self._device.sys_info["fade_off_time"]
+        elif (on_state := self.data.get("on_state")) and (
+            off_state := self.data.get("off_state")
+        ):
+            on_max = on_state.get("max_duration", self.MAXIMUM_DURATION)
+            off_max = off_state.get("max_duration", self.MAXIMUM_DURATION)
+            on_enabled = on_state["enable"]
+            off_enabled = off_state["enable"]
+            on_duration = on_state["duration"]
+            off_duration = off_state["duration"]
+        else:
             raise KasaException(
                 f"Unsupported for {self.REQUIRED_COMPONENT} v{self.supported_version}"
             )
 
-        return self.data["off_state"]
-
-    async def set_enabled_v1(self, enable: bool):
+        self._enabled = on_enabled or off_enabled
+        self._on_state = {
+            "duration": on_duration,
+            "enable": on_enabled,
+            "max_duration": on_max,
+        }
+        self._off_state = {
+            "duration": off_duration,
+            "enable": off_enabled,
+            "max_duration": off_max,
+        }
+
+    async def set_enabled(self, enable: bool):
         """Enable gradual on/off."""
-        return await self.call("set_on_off_gradually_info", {"enable": enable})
+        if not self._supports_on_and_off:
+            return await self.call("set_on_off_gradually_info", {"enable": enable})
+        else:
+            duration = self.DEFAULT_DURATION if enable else 0
+            on = await self.set_turn_on_transition(duration)
+            off = await self.set_turn_off_transition(duration)
+            return {**on, **off}
 
     @property
-    def enabled_v1(self) -> bool:
+    def enabled(self) -> bool:
         """Return True if gradual on/off is enabled."""
-        return bool(self.data["enable"])
+        return self._enabled
+
+    @property
+    def turn_on_transition_feature(self) -> Feature | None:
+        """Return inspectable details of the feature or None if unsupported."""
+        return self._module_features.get("smooth_transition_on")
```

Comment:
Do we really need these property getters considering the transition is not an interface, and these are already only accessed through the features-api?

Reply:
It saves having to add public properties for maximum and minimum values.  I think it's simpler to understand a common implementation approach for introspective properties like ranges and units rather than having only the interfaces expose the features.

Reply:
If it's just for tests, we could probably just access the feature directly (as our IDs are stable)? Sorry, I'm a bit exhausted due to this weather to think about this atm.

Reply:
That's removed now and using the feature directly.